lib/bsdlib/bsdlib.c

```diff
@@ -20,14 +20,26 @@ extern void ipc_proxy_irq_handler(void);
 
 static int _bsdlib_init(struct device *unused)
 {
+	int err;
+
 	/* Setup the network IRQ used by the BSD library.
-	 * Note: No call to irq_enable here, that is done through bsd_init.
+	 * Note: No call to irq_enable() here, that is done through bsd_init().
 	 */
 	IRQ_DIRECT_CONNECT(BSD_NETWORK_IRQ, BSD_NETWORK_IRQ_PRIORITY,
 			   ipc_proxy_irq_handler, 0);
-	bsd_init();
 
-	return 0;
+	err = bsd_init();
+
+	if (IS_ENABLED(CONFIG_BSD_LIBRARY_SYS_INIT)) {
```

Comment:
This was done like this to make sure I could still call `_bsdlib_init()` when `CONFIG_BSD_LIBRARY_SYS_INIT` isn't defined, to prevent a compiler warning (unused function).